radare2: add page (#3433)
tldr-pages/tldr: pages/common/radare2.md

# radare2

> A set of reverse engineering tools.
> More information: <https://radare.gitbooks.io/radare2book/>.

> - Open a file in write mode without parsing the file format headers.:

Reviewer: Perhaps this will read better?
```suggestion
- Open a file in write mode without parsing the file format headers:
```

> `radare2 -nw {{binary}}`

Reviewer: Perhaps this will read better? Same fro the examples below
```suggestion
`radare2 -nw {{path/to/binary}}`
```

- Debug a program:

`radare2 -d {{binary}}`

> - Run a script before entering interactive command-line prompt:

Reviewer: Perhaps this will read better?
```suggestion
- Run a script before entering the interactive command-line prompt:
```

`radare2 -i {{script.r2}} {{binary}}`

> - Show a help page of any radare2 command within the interactive client:

Reviewer: Perhaps this will read better?
```suggestion
- Show help text for any command in the interactive CLI:
```

`> {{radare2_command}}?`

> - Run a shell command from interactive radare:

Reviewer: Perhaps this will read better?
```suggestion
- Run a shell command from the interactive CLI:
```

Why is the `!` required here?

Author: Thank you for your review.

Since this command is within the interactive CLI, the user is not in a regular shell. So to be able to execute **shell** commands from the interactive CLI, the `!` is needed here.
For example if the user wants to look up a file within the CLI, he does not have to exit the CLI, but can just do the following:
`> !ls`
which will execute the shell command `ls`.

Reviewer: Ah, I see now. Thanks for the clarification :-)
